lib/PEX.ts

```diff
@@ -18,22 +23,39 @@ import {
   PresentationResult,
   PresentationSignCallBackParams,
   PresentationSubmissionLocation,
+  SdJwtDecodedVerifiableCredentialWithKbJwtInput,
+  SdJwtKbJwtInput,
   VerifiablePresentationFromOpts,
   VerifiablePresentationResult,
 } from './signing';
 import { DiscoveredVersion, IInternalPresentationDefinition, IPresentationDefinition, PEVersion, SSITypesBuilder } from './types';
-import { definitionVersionDiscovery, getSubjectIdsAsString } from './utils';
+import { calculateSdHash, definitionVersionDiscovery, getSubjectIdsAsString } from './utils';
 import { PresentationDefinitionV1VB, PresentationDefinitionV2VB, PresentationSubmissionVB, Validated, ValidationEngine } from './validation';
 
+export interface PEXOptions {
+  /**
+   * Hasher implementation, can be used for tasks such as decoding a compact SD-JWT VC to it's encoded variant.
```

Comment:
What is the exact reason. I do not immediately see why async and sync could not behave in the same way. When looking up the Hasher implementation a week ago, I noticed they have 2 types/interfaces. One basically being the async version the other the sync version. Is there a specific reason for it? Right now it is just one function, so why not have a type that is either one or the other using an OrPromise<> ?

Reply:
I made the difference as otherwise a lot of top-level, now-sync, API methods have to become asynchronous in PEX, which I wanted to be cautious of. 

If that's ok with you then i can combine them